dapp/src/services/apy.service.js

```diff
@@ -0,0 +1,35 @@
+import { apyDayOptions } from 'utils/constants'
+
+export default class ApyService {
+  async fetchApy() {
+    const dayResults = await Promise.all(
+      apyDayOptions.map(async (days) => {
+        let endpoint, varName
+        if (apyDayOptions.includes(days)) {
+          endpoint = `${process.env.APR_ANALYTICS_ENDPOINT}/${days}`
+          varName = `apy${days}`
+        } else {
+          throw new Error(`Unexpected days param: ${days}`)
+        }
+
+        try {
+          const response = await fetch(endpoint)
+          if (response.ok) {
```

Comment:
If response is not ok react-query function should throw an error: 
https://react-query.tanstack.com/guides/query-functions#handling-and-throwing-errors

This way react query knows to attempt to refetch the data